server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifier.java

```diff
@@ -90,7 +90,7 @@ public IndexMetadata verifyIndexMetadata(IndexMetadata indexMetadata, Version mi
         // Next we have to run this otherwise if we try to create IndexSettings
         // with broken settings it would fail in checkMappingsCompatibility
         newMetadata = archiveBrokenIndexSettings(newMetadata);
-        checkMappingsCompatibility(newMetadata);
+        newMetadata = checkMappingsCompatibility(newMetadata);
```

Comment:
Can we add a new method for doing the transformation? The "check" action to me implies nothing will change, it's just checking things.

Reply:
It's only used here, so I'll simply rename the method and update the comments.